### PR TITLE
Fixes #145: disabling deploy button if the repository is not from github

### LINKED ISSUE #145: disable deploy button if the repository is not from github.
### This is a
- [x] Bug report
- [ ] Feature request

### Current behavior

deploy button is showing if the  repository is not from GitHub
### Expected behavior

disable deploy button  if the  repository is not from GitHub because now yaydoc OAuth supports the only github
### Steps to reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

I have added to regular expression for github's repository. if it not from github then i'm disabling gh-pages deploy
## Related Issue

close #145 
## Motivation and Context

yaydoc outh currently supports githuh based repository. if the repository is from other than github error will rasie when we pushing documentation to github pages 
## How Has This Been Tested?

Tested in my local environment and screenshot of disabled deploy button after generating documentation from bitbucket repository is added
## Screenshots 
disable
## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: public/scripts/validation.js [hunk 1]
```diff
@@ -35,8 +35,16 @@
 
     setMessages: function (val) {
       messages = val;
-    }
+    },
 
+    isGitHTTPS: function (url) {
```
Comment (imujjwal96) [inline public/scripts/validation.js]: *Github

File: routes/index.js [hunk 2]
```diff
@@ -14,10 +15,17 @@ router.get('/download/:email/:uniqueId', function (req, res, next) {
 });
 
 router.get("/github", function (req, res, next) {
-  req.session.uniqueId = req.query.uniqueId;
-  req.session.email = req.query.email
-  req.session.gitURL = req.query.gitURL
-  next()
+  if (validation.isGitSSH(req.query.gitURL) || validation.isGitHTTPS(req.query.gitURL)) {
```
Comment (pri22296) [inline routes/index.js]: I have a doubt here. ssh url wouldn't work with webUI, so should we validate ssh urls too?

Comment (sch00lb0y) [inline routes/index.js]: now it is not working. but we can make it work by changing the deploy generator code a bit.I'll raise a separate issue for this.

Comment (imujjwal96) [inline routes/index.js]: Is it possible to work with SSH URL without the necessary keys?

Comment (sch00lb0y) [inline routes/index.js]: from ssh we'll take username and repo name. we'll push to gh-pages with access token in https.

Comment (imujjwal96) [inline routes/index.js]: Why?

Comment (pri22296) [inline routes/index.js]: @sch00lb0y No need for that. we can't use ssh with webUI. just remove validation for SSH url!

File: public/scripts/validation.js [hunk 3]
```diff
@@ -35,8 +35,12 @@
 
     setMessages: function (val) {
       messages = val;
-    }
+    },
 
+    isGithubHTTPS: function (url) {
+      regex = /^https:\/\/github.com\/[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}\/[a-z\d-]{0,38}(\.git)?/
```
Comment (imujjwal96) [inline public/scripts/validation.js]: What will happen if I use HTTP instead of HTTPS?

Comment (sch00lb0y) [inline public/scripts/validation.js]: I have changed to support http also. Thanks !

**Comment by mariobehling** [comment]
Very good!

**Pull request merged**